src/main/java/seedu/address/model/person/GroupContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,26 @@
+package seedu.address.model.person;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person} belongs to a specific {@code Group}.
+ */
+public class GroupContainsKeywordsPredicate implements Predicate<Person> {
+    private final Group group;
+
+    public GroupContainsKeywordsPredicate(Group group) {
+        this.group = group;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return group.containsExact(person);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof GroupContainsKeywordsPredicate // instanceof handles nulls
```

Comment:
Would it be a good idea to change it to:
```
if (other == this)...

if (!(other instanceof GroupContainsKeywordsPredicate)) ...

Type casting, check equal
```
to improve readability?

Reply:
Great point! Thank you for pointing that out 👍